pkg/kudoctl/cmd/root.go

```diff
@@ -29,6 +29,7 @@ func NewKudoctlCmd() *cobra.Command {
 		Long: `KUDO CLI and future sub-commands can be used to manipulate, inspect and troubleshoot KUDO-specific CRDs
 and serves as an API aggregation layer.
 `,
+		SilenceUsage: true,
```

Comment:
So this means we'll silence it for all errors, not just syntax errors. I wonder if there's a way to selectively prevent from printing out usage just for some errors 🤔

Reply:
FWIK cobra will not silence usage for all errors, only those related to `RunE` functions (this could be a validation error),  for example [this command](https://github.com/kudobuilder/kudo/blob/master/pkg/kudoctl/cmd/init.go#L78-L83)
Cobra will still print usage if requested (`-h` flag is passed) or there is a syntax error.

Reply:
oh I see, I think we're still returning error through runE for the non-syntax related errors, should we maybe stop doing that? 🤔 

I need to think about this more, I think your solution is correct, I am just wondering if we're not suppressing the usage "too much" now as I think printing it automatically after syntax error is very useful.

Reply:
Yep, but I'd be ok with that. I like my "usage help" when I ask for it ;)

Reply:
@zen-dog yep "usage help" will be printed only if you pass and unknown flag or call it using "-h"

@alenkacz exactly ! actually in kudoctl, RunE functions return errors for syntax and non-syntax related errors. 
> should we maybe stop doing that? 🤔

In my opinion the best thing to do here is use `RunE` for handling execution errors and `Args` for handling syntax/args validation errors. `command.Args` is used to customise argument validation in cobra ( [docs here](https://github.com/spf13/cobra#positional-and-custom-arguments) )

for example `kubectl init` command can be rewritten like this

```go
	cmd := &cobra.Command{
		Use:     "init",
		Short:   "Initialize KUDO on both the client and server",
		Long:    initDesc,
		Example: initExample,
		Args: func(cmd *cobra.Command, args []string) error {
			if len(args) != 0 {
				return errors.New("this command does not accept arguments")
			}
			return i.validate(cmd.Flags());
		},
		RunE: func(cmd *cobra.Command, args []string) error {
			i.home = Settings.Home
			clog.V(8).Printf("init cmd %v", i)
			return i.run()
		},
	}
```

But i'm still not sure if flags validation should be handled in the `Args` function in this example

Reply:
It's worth experimenting with flags validation in the Args fn! TIL. Let's merge this and we can open an issue to try this out on one command and then issues to expand this out to every other command, wdyt @A-Hilaly?

Reply:
@gerred LGTM 👍 i would be glad to help in this !